crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,308 @@
+pub enum Advisory {
+    /// Useful context without action needed
+    Information,
+    /// Suggested change that would improve things
+    Improvement,
+    /// Something that breaks compliance rules
+    Violation,
+}
+
+/// Represents a health check advice
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Advice {
+    /// The key of the advice e.g. "is_deprecated"
+    pub key: String,
+    /// The value of the advice e.g. "true"
+    pub value: Value,
+    /// The message of the advice e.g. "This attribute is deprecated"
+    pub message: String,
+    /// The advisory of the advice e.g. "violation"
+    pub advisory: Advisory,
+}
+
+/// Provides advice on a sample attribute
+pub trait Advisor {
+    /// Provide advice on a sample attribute
+    fn advise(
+        &self,
+        attribute: &SampleAttribute,
+        health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice>;
+
+    //TODO conclude(&self) -> Option<Vec<Advice>>;
+    // Provide an overall summary of the advice e.g. LengthyAttributeNameAdvisor
+    // could provide statistics on the length of the attribute names: min, max, avg
+    // Each statistic would be an Advice with a key like "min_length", "max_length", "avg_length"
+}
+
+/// An advisor that checks if an attribute is deprecated
+pub struct DeprecatedAdvisor;
+impl Advisor for DeprecatedAdvisor {
+    fn advise(
+        &self,
+        _attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        if let Some(attribute) = semconv_attribute {
+            if attribute.deprecated.is_some() {
+                Some(Advice {
+                    key: "is_deprecated".to_owned(),
+                    value: Value::Bool(true),
+                    message: "Is deprecated".to_owned(),
+                    advisory: Advisory::Violation,
+                })
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that checks if an attribute is stable from the stability field in the semantic convention
+/// The value will be the stability level
+pub struct StabilityAdvisor;
+// TODO: Configurable Advisory level, strictly stable would mean Violation
+
+impl Advisor for StabilityAdvisor {
+    fn advise(
+        &self,
+        _attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        if let Some(attribute) = semconv_attribute {
+            match attribute.stability {
+                Some(ref stability) if *stability != Stability::Stable => Some(Advice {
+                    key: "stability".to_owned(),
+                    value: Value::String(stability.to_string()),
+                    message: "Is not stable".to_owned(),
+                    advisory: Advisory::Improvement,
```

Comment:
How do we decide what constitutes an improvement versus a violation? For instance, this `StabilityAdvisor` generates an improvement when the attribute isn't `Stable`. However, in some scenarios, a user might consider this a violation. Perhaps there's a mechanism elsewhere in the system to handle this in one way or another.

Reply:
I have a task in the PR to add a configuration file. This will allow you to:
- select which advisors you want
- configure those advisors
At the moment, both the Enum and Stability advisors need configurable advisory level. I've chosen "sensible" defaults.